src/Agent.Listener/Configuration/AutoLogonConfigurationManager.cs

```diff
@@ -0,0 +1,283 @@
+        }
+
+        private void UpdateRegistriesForAutoLogon(AutoLogonRegistryManager regHelper, string userName, string domainName, string logonPassword)
+        {
+            regHelper.UpdateStandardRegistrySettings();
+
+            //auto logon
+            ConfigureAutoLogon(regHelper, userName, domainName, logonPassword);
+
+            //startup process
+            var startupProcessPath = Path.Combine(HostContext.GetDirectory(WellKnownDirectory.Bin), "agent.listener.exe");
+            var startupCommand = string.Format($@"{startupProcessPath} run");
+            Trace.Verbose($"Setting startup command as {startupCommand}");
+
+            regHelper.SetStartupProcessCommand(startupCommand);
+        }
+
+        private void ConfigureAutoLogon(AutoLogonRegistryManager regHelper, string userName, string domainName, string logonPassword)
+        {
+            //find out if the autologon was already enabled, show warning in that case
+            ShowAutoLogonWarningIfAlreadyEnabled(regHelper, userName);
+
+            var windowsHelper = HostContext.GetService<INativeWindowsServiceHelper>();
+            windowsHelper.SetAutoLogonPassword(logonPassword);
+
+            regHelper.UpdateAutoLogonSettings(userName, domainName);
+        }
+
+        private void ShowAutoLogonWarningIfAlreadyEnabled(AutoLogonRegistryManager regHelper, string userName)
+        {
+            regHelper.FetchAutoLogonUserDetails(out string autoLogonUserName, out string domainName);
+            if(autoLogonUserName != null && !userName.Equals(autoLogonUserName, StringComparison.CurrentCultureIgnoreCase))
+            {
+                _terminal.WriteLine(string.Format(StringUtil.Loc("AutoLogonAlreadyEnabledWarning"), userName));
+            }
+        }
+
+        private bool IsCurrentUserSameAsAutoLogonUser()
+        {
+            var regHelper = new AutoLogonRegistryManager(HostContext.GetService<IWindowsRegistryManager>());            
+            regHelper.FetchAutoLogonUserDetails(out string userName, out string domainName);
+            if(string.IsNullOrEmpty(userName) || string.IsNullOrEmpty(domainName))
+            {
+                throw new InvalidOperationException("AutoLogon is not configured on the machine.");
+            }
+
+            var nativeWindowsHelper = HostContext.GetService<INativeWindowsServiceHelper>();
+            return nativeWindowsHelper.HasActiveSession(domainName, userName);
+        }
+
+        private void DisplayWarningsIfAny(AutoLogonRegistryManager regHelper)
+        {
+            var warningReasons = regHelper.GetAutoLogonRelatedWarningsIfAny();
+            if(warningReasons.Count > 0)
+            {
+                WriteSection(StringUtil.Loc("UITestingWarning"));
+                for(int i=0; i < warningReasons.Count; i++)
+                {
+                    _terminal.WriteLine(String.Format("{0} - {1}", (i+1).ToString(), warningReasons[i]));
```

Comment:
pass CultureInfo.CurrentCulture. don't call tostring on the integer, string.format will call tostring for you.